src/popup/components/NameItem.vue

```diff
@@ -163,76 +174,92 @@ export default {
     address: { type: String, default: '' },
     autoExtend: { type: Boolean },
   },
-  data: () => ({
-    expand: false,
-    newPointer: '',
-    showInput: false,
-    error: false,
-    UNFINISHED_FEATURES: process.env.UNFINISHED_FEATURES,
-  }),
-  subscriptions() {
-    return pick(this.$store.state.observables, ['topBlockHeight']);
-  },
-  computed: {
-    ...mapGetters(['account']),
-    nameEntry() {
-      return this.$store.getters['names/get'](this.name);
-    },
-    hasPointer() {
-      return this.nameEntry?.pointers?.accountPubkey;
-    },
-    addressOrFirstPointer() {
-      return this.nameEntry?.pointers?.accountPubkey
-        || Object.values(this.nameEntry?.pointers || {})[0];
-    },
-  },
-  watch: {
-    newPointer() {
-      this.error = false;
-    },
-  },
-  methods: {
-    blocksToRelativeTime,
-    checkAddressOrChannel,
-    async insertValueFromClipboard() {
-      this.newPointer = await readValueFromClipboard();
-    },
-    async setDefault() {
-      await this.$store.dispatch('names/setDefault', {
-        address: this.account.address,
-        name: this.name,
+  setup(props, { root }) {
+    const expand = ref(false);
+    const newPointer = ref<string | undefined>('');
+    const showInput = ref(false);
+    const error = ref(false);
+
+    const account = useGetter<IAccount>('account');
+
+    const nameEntry = computed<IName | null>(() => root.$store.getters['names/get'](props.name));
+    const hasPointer = computed(() => nameEntry.value?.pointers?.accountPubkey);
+    const addressOrFirstPointer = computed((): string | null => (
+      nameEntry.value?.pointers?.accountPubkey
+      || Object.values(nameEntry.value?.pointers || {})[0]
+    ));
+
+    const topBlockHeight = rxJsObservableToVueState<number>(
+      (root.$store.state as any).observables.topBlockHeight,
+    );
+
+    async function insertValueFromClipboard() {
+      newPointer.value = await readValueFromClipboard();
```

Comment:
```suggestion
      newPointer.value = (await readValueFromClipboard()) || ''; 
```
then you can change type to string from string | undefined in ref

Reply:
OK